### ISSUE TITLE
Release train for 1.0

### ISSUE THREAD

**Issue by astubbs** [confluentinc/parallel-consumer#172]
- [x] README cleanup/update regarding maintenance 
- [x] README regarding “alpha” state
- [x] offset documentation
- [ ] javadoc cleanup sweep
- [ ] dead code sweep
- [ ] todo sweep
- [ ] disabled test sweep
- [ ] potentially remove streaming interfaces

Issues targeted for 0.5:
----

In priority order - only items marked *high* will block the 1.0 release.

Tech debt / refactoring:
- [x] #223
- [x] #224 
- [x] #218
- [x] #219 
 
Tech debt:
- [x] #201
- [x] #236
- [x] #229
- [x] #197
- [x] #215

Enhancements High:
- [x] #216 

Features High: 
- [x] #18

Features Low:

Issues targeted for NEXT:
----

Fixes:
- [x] #205
- [x] #203
- [x] #184
- [x] #195
- [x] #318

Tech debt:
- [ ] #271
- [x] #289
- [ ] #270
- [ ] #220
- [ ] #192
- [ ] #290 

Enhancements High:
- [x] #268 
- [x] #242
- [ ] #78
- [x] #65
- [x] #297 
- [x] #327

Features High: 

Features Low:
- [x] #193
- [x] #110

Issues targeted for NEXT NEXT:
----

Features:
- [x] #175
- [ ] #170
- [ ] #71
- [x] #187
- [x] #183
- [x] #191
- [ ] #186
- [x] #185
- [x] #176
- [ ] #310
- [x] #372

Tech debt:
- [ ] #259
- [x] #247
- [x] #246
- [ ] #81

Issues targeted for post 1.0:
----

Features high:
- [x] #232
- [x] #226
- [x] #225
- [ ] #241

Examples Low:
- [ ] #178
- [ ] #171

Tech debt:
- [ ] #233
- [ ] #103
- [ ] #200
- [x] #144 

Features low:
- [ ] #196

**Comment by sribee** [comment]
@astubbs - This is a very interesting and useful project for improving the kafka consumer performance and also to reduce the load on the broker. Is there a timeline you could provide for the 1.0 release? thanks in advance.

**Comment by astubbs** [comment]
Yes, will be Q1 22. Once I finish up some work on another project that I'm on, I'm coming back to PC to finish up some things and do the release.

It should be noted, the only changes for 1.0 will be a couple new features, some tidy up, tighten up some of the API and things. There are no known bugs or issues that would be fixed.

**Comment by astubbs** [comment]
Set target issues in priority order. Only high priority items will block 1.0

**Comment by raphaelauv** [comment]
#18 is in high priority , cool !

**Comment by astubbs** [comment]
> #18 is in high priority , that's cool !

Batching is now drafted complete, and awaiting merge early next week! https://github.com/confluentinc/parallel-consumer/issues/18
I know I know, it's been waiting a while - thanks for the help @1123 !

**Comment by astubbs** [comment]
> > #18 is in high priority , that's cool !
> 
> Batching is now drafted complete, and awaiting merge early next week!

Merged!

**Comment by astubbs** [comment]
FYI 0.5 is out

**Comment by raphaelauv** [comment]
scikit-learn have been on v0 for more than 12 years :grin:

**Comment by astubbs** [comment]
Tracked in the maintained fork as astubbs/parallel-consumer#135, which carries a summary of this
issue and its current status there.

The **Confluent-hosted version** of this project is no longer maintained - its README says so.
**Parallel Consumer itself continues** in my fork: https://github.com/astubbs/parallel-consumer

Nothing in this repository is closed or changed by this comment.
